term/fhir-term/src/main/java/com/ibm/fhir/term/util/CodeSystemSupport.java

```diff
@@ -232,6 +247,124 @@ public static CodeSystem getCodeSystem(java.lang.String url) {
         return FHIRRegistry.getInstance().getResource(url, CodeSystem.class);
     }
 
+    /**
+     * @param codeSystem
+     * @return
+     *    true if the passed codeSystem has a property-based concept hierarchy
+     */
+    public static boolean hasPropertyHierarchy(CodeSystem codeSystem) {
+        for (CodeSystem.Property p : codeSystem.getProperty()) {
```

Comment:
how many properties do we have to run through each time?

Reply:
usually just a couple.  the most i've seen is around 10:  https://terminology.hl7.org/3.1.0/CodeSystem-v3-ActCode.json.html

Reply:
OK. Hopefully we won't be hitting this often enough to impact performance.